proto/trip.proto

```diff
@@ -236,8 +236,12 @@ message TripLeg {
     }
     repeated Level levels = 61;
     uint32 level_precision = 62;
-    Speeds speeds_faded = 63;
-    Speeds speeds_non_faded = 64;
+    oneof has_speeds_faded {
```

Comment:
Maybe instead of introducing this strange syntax it is possible to follow approach introduced in https://github.com/valhalla/valhalla/pull/3527 - just consider the default value as "not set"?

Reply:
Anyway, speed `0` is considered everywhere as "not set" or even "unknown".

Reply:
What's the default of a message? Surely not 0. For the primitive speed values you're right.

Reply:
stilll, if all properties of `Speeds` message are zeroes, is it enough to consider message as not set?

Reply:
sure, but I very much doubt that this would translate to `speeds_faded() == false` in protobuf. I rather use this awful oneof than introducing a new function checking every field instead.